src/lib/ReactCore_Internal.re

```diff
@@ -11,7 +13,7 @@ module Make = (Implementation: HostImplementation) => {
     let reset = () => {
       debug := true;
       componentKeyCounter := 0;
-      instanceIdCounter := 1 /* id 0 reserved for defaultStringInstance */
```

Comment:
I don't think we have `defaultStringInstance`. We can remove this comment and set `instanceIdCounter` to 0.